src/textual/renderables/text_opacity.py

```diff
@@ -62,21 +62,24 @@ def process_segments(
         _Segment = Segment
         _from_color = Style.from_color
         if opacity == 0:
-            for text, style, control in segments:
+            for text, style, control in cast(
+                Iterable[tuple[str, Style, object]], segments
+            ):
+                assert style is not None
                 invisible_style = _from_color(bgcolor=style.bgcolor)
                 yield _Segment(cell_len(text) * " ", invisible_style)
         else:
             for segment in segments:
-                text, style, control = segment
-                if not style:
+                text, maybe_style, control = segment
```

Comment:
I don't think it is a maybe style. It is a `Style` at this point. Styles do have `__bool__` to indicate a null style, but it is still a Style object.

Reply:
It is still a maybe_style because in this branch I haven't cast segments to indicate it can't be None. I think you're saying I should do that in both branches. Will do.